app/controllers/admin/ex_links_controller.rb

```diff
@@ -0,0 +1,146 @@
+# encoding:UTF-8
+class Admin::ExLinksController < ApplicationController
+  load_permissions_and_authorize_resource
+  before_action :authorize
+  before_action :set_ex_link, only: [:show, :edit, :update, :destroy]
+
+  # GET /ex_links
+  def index
+    if params[:tag]
+      @ex_links = get_exlinks_if_all_tags_present(params[:tag])
+    else
+      @ex_links = ExLink.all
+    end
+    @tags = get_tags
+  end
+
+  # GET /ex_links/1
+  def show
```

Comment:
I have changed a lot on this topic, but it is worth being more explicit and writing:
`@ex_link = ExLink.find(params[:id])`